flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/debezium/reader/SnapshotSplitReader.java

```diff
@@ -403,4 +403,58 @@ public boolean isRunning() {
             return currentTaskRunning;
         }
     }
+
+    private boolean containsPrimaryKey() {
+        return !CollectionUtil.isNullOrEmpty(currentSnapshotSplit.getSplitKeyType().getFields());
```

Comment:
It is better not to judge in this way. The chunk key will be set when we have to handle the big table in some cases to provide the at-least-once guarantee.

Reply:
We could init the SnapshotRecordCollector when the `record.key()` returns null.

Reply:
> It is better not to judge in this way. The chunk key will be set when we have to handle the big table in some cases to provide the at-least-once guarantee.

Thanks for your review. You mean that the chunk key may be set even though the table has no primary keys?

Reply:
Yes. If the table contains too much data, there is a performance problem with this solution.
In this case, we may provide the at-least-once semantics instead of the exactly once semantics.

Reply:
Is it possible that the exactly once semantic is required in some scenarios ? How about to provide an option to make it optional for users who can choose between better performance or exactly once semantic?

Reply:
I think we should control this by the experimental option `scan.incremental.snapshot.chunk.key-column`.
By default, we will return a single split. If users set the `scan.incremental.snapshot.chunk.key-column`, we will split the table into multi splits.